src/KSVD.jl

```diff
@@ -152,18 +152,9 @@ function ksvd(Y::AbstractMatrix{T}, n_atoms::Int, max_nnz=max(3, n_atoms ÷ 100)
     termination_condition = :nothing
     tic = time()
     for iter in 1:maxiters
-        verbose && @info "Starting svd"
-        (Y_, X_) = if !isnothing(minibatch_size)
-            minibatch_indices = sort(shuffle(axes(Y, 2))[1:minibatch_size])
-            Y_ .= Y[:, minibatch_indices]
-            X_ = X[:, minibatch_indices]
-            (Y_, X_)
-        else
-            (Y, X)
-        end
-        ksvd_update(ksvd_update_method, Y_, D, X_; timer)
-        verbose && @info "Starting sparse coding"
-        X = sparse_coding(sparse_coding_method, Y, D; timer)
+        # note that D gets updated in place
+        X = ksvd_loop!(ksvd_loop_type, ksvd_update_method, sparse_coding_method,
```

Comment:
Looks like currently the minibatch functionality is lost. Perhaps think about this some more...

Reply:
Reimplemented :+1: